Add documentation for a catching invalid index type

pavanky opened the pull request:
Summary:
The assert in in compute_q8gemm_prepacked_sparse_dq is currently unreachable.

Added inline comments to explain what is happening.

Test Plan: Ran  qnnpack q8gemm-sparse-test to verify.

Differential Revision: D43930667



cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

salilsdesai reviewed (approved):
Looks good (y). Thanks

pavanky committed:
Add documentation for a catching invalid index type (#96451)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/96451

The assert in in compute_q8gemm_prepacked_sparse_dq is currently unreachable.

Added inline comments to explain what is happening.

Test Plan: buck run caffe2/aten/src/ATen/native/quantized/cpu/qnnpack:q8gemm-sparse-test

Reviewed By: salilsdesai

Differential Revision: D43930667

fbshipit-source-id: 0e90b03e5b181bb41669ccab02a642c51146d5ec
